app/src/forms/proxy/service.js

```diff
@@ -0,0 +1,118 @@
+const { encryptionService } = require('../../components/encryptionService');
+const { ExternalAPI } = require('../../forms/common/models');
+
+const headerValue = (headers, key) => {
+  if (headers && key) {
+    try {
+      return headers[key.toUpperCase()] || headers[key.toLowerCase()];
+    } catch (error) {
+      return null;
+    }
+  } else {
+    throw new Error('Headers or Header Name not provided.');
+  }
+};
+
+const trimLeadingSlashes = (str) => str.replace(/^\/+|\$/g, '');
+const trimTrailingSlashes = (str) => str.replace(/\/+$/g, '');
+
+const service = {
+  generateProxyHeaders: async (payload, currentUser, token) => {
```

Comment:
I think that there is a replay problem here, as the token has no expiry. If someone was to capture a token it could be reused forever and have the victim's authorizations.

Reply:
We are only passing the token to external apis that we trust and authorized. Tokens don’t have expiry dates, you check the issued date and decide what time frame you accept.

Reply:
I guess if you’re talking about the proxy headers we generate only the chefs backend understands how to decrypt that data. And the front end stores it for that page only. So it’s ephemeral. We could add an issued time and reject if it’s older than whatever our timeframe is. Of course we then limit the use cases for external api to pretty much form load time only.

Reply:
Yes, sorry, too many "tokens" - I was meaning the proxy headers. Even though only the backend can decrypt it, the backend has no idea if the proxy headers object is recent or years old.

I wonder if we can just add a timestamp to the proxy headers object and let the third-party API decide what "too old" means to them.

Reply:
interesting... i like that - add a timestamp to know when this user info was gathered. great idea! will put that in.

Reply:
Ok, covered off a few things related to this.

So adding headers with values straight from the token:

- `X-CHEFS-TOKEN-IAT` : token `iat` - token issued 
-` X-CHEFS-TOKEN-EXP` : `exp` - token expiry, between expiry and issued at the external API can determine their own valid timeframe
- `X-CHEFS-TOKEN-SUB` : `sub` - @robinwn wanted this so including it, subject from IDP provider

I like your idea of including the timestamp and leaving it to external API to do as they see fit.